### DISCUSSION TITLE
XML blocks

### DISCUSSION THREAD

**Discussion by lego7set** [TurboWarp/extensions#1039] [Q&A]
What can I do with XML blocks? Basically all I know is that you can add hidden existing blocks into the block palette or pre-add blocks into custom c-blocks. I dont know what else you can do with it. Like can you change the shape of a block? Can you create inline blocks from penguin mod (i dont think so). 

So basically I found this
 ```
<block type="control_inline_stack_output">
    <value name="SUBSTACK">
        <block type="procedures_return">
            <value name="return">
            	<shadow type="text">
            		<field name="TEXT">1</field>
            	</shadow>
            </value>
        </block>
    </value>
</block>
``` 
from penguin mod (a mod of turbowarp) and its supposed to add an inline block or smth but i got bored. So I modified it (I know HTML and this looks like HTML so yeah) to make this

```
(function(Scratch) {
  'use strict';

  if (!Scratch.extensions.unsandboxed) {
    throw new Error('This Hello World example must run unsandboxed');
  }

  class HelloWorld {
    getInfo() {
      return {
        id: 'helloworldunsandboxed',
        name: 'Unsandboxed Hello World',
        blocks: [
          {
            opcode: 'hello',
            blockType: Scratch.BlockType.XML,
            xml: `<block type="control_for_each">
    <value name="SUBSTACK">
        <block type="procedures_return">
            <value name="return">
            	<shadow type="text">
            		<field name="TEXT">1</field>
            	</shadow>
            </value>
        </block>
        <block type = "motion_movesteps">
        </block>
    </value>
</block>`
          },
          {
             opcode: "inline",
             blockType:Scratch.BlockType.REPORTER,
             text: "Inline Blocks"
          }
        ]
      };
    }
    inline() {
        console.log("Used!")
        return 5
    }
  }
  Scratch.extensions.register(new HelloWorld());
})(Scratch);
```

So how the heck are you supposed to use this for? and why is this blocktype even available?? oh and also the inline block was because i wanted to see what would happen if i added a substack to a reporter.

**Comment by CST1229** [comment]
The XML BlockType lets you insert anything into the exttension's block palette. For example, All Menus uses it.

**Reply by lego7set** [reply]
Is there anything else it can do?
